src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -1457,6 +1459,34 @@ else if (0 == requestedPacketSize)
 					SQLServerException.makeFromDriverError(this, this, form.format(msgArgs), null, false);
 				}
 			}
+			
+			sPropKey = SQLServerDriverIntProperty.QUERY_TIMEOUT.toString();
+			int defaultQueryTimeout = SQLServerDriverIntProperty.QUERY_TIMEOUT.getDefaultValue();
+			queryTimeoutSeconds  = defaultQueryTimeout; //Wait forever
+			if (activeConnectionProperties.getProperty(sPropKey) != null  && 
+					activeConnectionProperties.getProperty(sPropKey).length() > 0)
+			{
+				try
+				{
+					int n = (new Integer(activeConnectionProperties.getProperty(sPropKey))).intValue();
+					if (n>=defaultTimeOut){
+						queryTimeoutSeconds = n;
+					}
+					else
+					{
+						MessageFormat form = new MessageFormat(SQLServerException.getErrString("R_invalidQueryTimeout"));
+						Object[] msgArgs = {activeConnectionProperties.getProperty(sPropKey)};
```

Comment:
I think in case of wrong querytimeout we should fallback to some valid querytimeout by logging error message.